mobile/src/main/java/ch/epfl/sdp/mobile/ui/prepare_game/PrepareGameScreenState.kt

```diff
@@ -1,23 +1,31 @@
 package ch.epfl.sdp.mobile.ui.prepare_game
 
+import ch.epfl.sdp.mobile.application.authentication.AuthenticatedUser
+import ch.epfl.sdp.mobile.application.chess.ChessFacade
+import ch.epfl.sdp.mobile.application.chess.Match
+import ch.epfl.sdp.mobile.ui.social.Person
+import kotlinx.coroutines.CoroutineScope
+
 /**
  * State interface of the [PrepareGameScreen]
  * @property colorChoice chosen color
- * @property gameType chosen game type
+ * @property selectedOpponent the selected opponent whom to create a game with
  */
-interface PrepareGameScreenState {
-  var colorChoice: ColorChoice
-  var gameType: GameType
-}
+interface PrepareGameScreenState<P : Person> {
 
-/** Color choices for a chess game */
-enum class ColorChoice {
-  White,
-  Black
-}
+  /** Color choices for a chess game */
+  enum class ColorChoice {
+    White,
+    Black
+  }
 
-/** Game types for local and online chess games */
-enum class GameType {
-  Local,
-  Online
+  var colorChoice: ColorChoice
+  val opponents: List<P>
+  var selectedOpponent: P?
+  val user: AuthenticatedUser
+  val navigateToGame: (Match) -> Unit
+  val onPlayClick: (P) -> Unit
+  val onCancelClick: () -> Unit
+  val scope: CoroutineScope
+  val chessFacade: ChessFacade
```

Comment:
The `ChessFacade` should not be exposed in the `PrepareGameScreenState` interface.

Reply:
Fixed in [081ef90](https://github.com/epfl-SDP/android/pull/200/commits/081ef9062dc1f7891b3ae7064430cd8d60880299)